### ISSUE TITLE
Seattle.rb style no longer valid in Ruby 2.4.2 when curly braces are used.

### ISSUE THREAD

**Issue by danielpclark** [rubocop-hq/rubocop#4793]
As of Ruby 2.4.2 there is a breaking change for any Ruby code that doesn't use parenthesis on a literal before curly braces.

This is no longer allowed:

```ruby
define_method :some_method_name { "asdf" }
```
This must now be either:

```ruby
define_method(:some_method_name) { "asdf" }
```

or 

```ruby
define_method :some_method_name do "asdf" end
```
According to Ruby Bug reports https://bugs.ruby-lang.org/issues/13939 and https://bugs.ruby-lang.org/issues/13898 this is the way it has to be.

>  A brace block has higher precedence and is bound to the previous expression, and a literal cannot be a method call and have a block. - Nobuyoshi Nakada

--------

## Expected behavior

When Rubocop finds any literal followed by curly braces suggest wrapping the literal in parenthesis as a parameter or substituting the curly braces for `do` and `end`.

Anyone planning to upgrade their project to Ruby 2.4.2 or higher may want to have this pointed out with a cop.

**Comment by Drenmi** [comment]
This is covered by `Lint/AmbiguousBlockAssociation`. 🙂

> This cop checks for ambiguous block association with method when param passed without parentheses.

Good to see that Ruby is introducing its own check.

**Comment by bbatsov** [comment]
I'm closing this ticket due to no recent activity. Feel free to re-open it if you ever come back to it.
